Merge Conflict/Assets/Scripts/ComponentHandler.cs

```diff
@@ -155,8 +165,82 @@ private void HandleOverlappingObjects()
+#nullable enable
+            if (Tags.SellingStation.UsedByGameObject(staticComponent.gameObject))
+            {
+
+                // TODO: change this ("requiredQuestComponent") later to the correct "requiredQuestComponent" from actual quest
+                // GameObject requiredQuestComponent = Components.HDD.InstantiateGameObjectAndAddTexture(new Vector2(300, 400));
+                // GameObject requiredQuestComponent = Components.CreateCase().InstantiateGameObjectAndAddTexture(new Vector2(300, 400));
+                // GameObject requiredQuestComponent = Components.CreateCase(powersupply: null, hdd: Components.HDD, motherboard: null).InstantiateGameObjectAndAddTexture(new Vector2(300, 400));
+                GameObject requiredQuestComponent = Components.CreateCase(
+                    powersupply: null,
+                    hdd: Components.HDD.Clone(),
+                    motherboard: Components.CreateMB(
+                            cpu: null,
+                            ram: Components.RAM.Clone(),
+                            gpu: Components.GPU.Clone()
+                        )
+                    ).InstantiateGameObjectAndAddTexture(new Vector2(300, 400));
+
+                Element? requiredQuestElement
+                    = requiredQuestComponent.TryGetComponent(out ComponentHandler requiredQuestComponentHandler)
+                    ? requiredQuestComponentHandler.element
+                    : null;
+
+                Element? draggedElement
+                    = draggedComponent.TryGetComponent(out ComponentHandler draggedComponentHandler)
+                    ? draggedComponentHandler.element
+                    : null;
+
+                if (draggedElement == null)
+                {
+                    return;
+                }
+
+                if (draggedElement.IsEqual(requiredQuestElement))
+                {
+
+                    //TODO: call xp/money controller instead  ->  give more xp/money than putting it in trashcan  =>  use salesValues of the components                    
+                    int actualSalesValue = draggedElement.GetSalesValue();
+                    Debugger.LogMessage($"salesValue : {actualSalesValue}".ToString());
+
+                    Destroy(draggedComponent, timeToDestroyObject);
+                    Debugger.LogMessage("Component was sold. Congratulations! You have completed a quest.");
+                }
+                else
+                {
+                    Debugger.LogMessage("Component cannot be sold. It does not correspond to the required order from the quest.");
+                    // if component cannot be sold -> automatically move it back onto the playfield
+                    draggedComponent.GetComponent<ComponentHandler>().MoveComponent(new Vector2(200, 400), 100f);
+                }
+
             }
+#nullable restore
+        }
+    }
+
+    public void MoveComponent(Vector2 targetPosition, float movingSpeed)
```

Comment:
Movement should be handled in ComponentMovement Script and follow the established workflow. Maybe contact @Semeseen for this?

Reply:
Yes, ehm, the movement is then also included in the MC-37 ticket. I merge it first so that the pull request is ready and then adjust the movement afterwards.